examples/custom_queue_example/customqueuemain.cpp

```diff
@@ -0,0 +1,47 @@
+#include <farbot/fifo.hpp>
+#include <rtlog/rtlog.h>
+
+template <typename T> class FarbotMPSCQueueWrapper {
```

Comment:
@atsushieno I incorporated your work here as an example for future readers. Would you please review this?

I'll make sure you get marked as a co-author in the git commit, as I took much of this verbatim from your branch :)

Reply:
Yes everything looks much greater with the customizible queues! I'm glad my changes were of some help.

I don't try hard to push my MPSC stuff in the main code from "example use case", but what I'm concerned however is that people would like to have default implementation for MPSC anyway, and they would simply ask you to incorporate that farbot version (or anything equivalent) after all, just like there is the original SPSC implementation.
For my own purpose I will keep my friendly fork version (maybe not even a fork, just an addon header would work), and I'm a bit concerned that it may be more widely referenced than your original repo as people would usually want to use the logging functionality in multi-threaded apps.
As a library author in this world I understand that I wouldn't like to incorporate external deps that might not match my taste (or being asked to do so), but having third-party fork more popular (especially widely used without my fixes) would not be a great experience.
IF that were the case in the future, I'm always happy to share my stuff and feel free to take them back into this repo.

Reply:
That makes sense.

How about this: I submit this change, then either myself or you (if you had time and wanted to) "swap" farbot and readerwriterqueue. Farbot becomes the main dependency, we add two versions of the wrapper - one single producer and one multi-producer. readerwriterqueue becomes the example dependency.

Overall, my only concern with incorporating farbot AND readerwriterqueue together is that we would be downloading more dependencies to user's machines. I already feel partly bad downloading stb, libfmt AND readerwriterqueue. Most libraries I know and trust don't download anything and are "self reliant". I would like to not add additional dependencies to the mix if possible. I agree that it is reasonable for users to want both single producer and multi producer versions of rtlog.

If we swapped in farbot as the primary queue, this would keep the dependency count the same and add more flexibility for the user. It would also be easy for anyone to add a new queue on their own if they wanted it.

What do you think?

Reply:
Sounds good. I'll ping you in any reviews if I take on the work, please do the same for me!